store/sqlstore/thread_store.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/store"
+	"github.com/pkg/errors"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+type SqlThreadStore struct {
+	SqlStore
+}
+
+func (s *SqlThreadStore) ClearCaches() {
+}
+
+func newSqlThreadStore(sqlStore SqlStore) store.ThreadStore {
+	s := &SqlThreadStore{
+		SqlStore: sqlStore,
+	}
+
+	for _, db := range sqlStore.GetAllConns() {
+		table := db.AddTableWithName(model.Thread{}, "Threads").SetKeys(false, "PostId")
+		table.ColMap("PostId").SetMaxSize(26)
+		table.ColMap("Participants").SetMaxSize(0)
+	}
+
+	return s
+}
+
+func threadSliceColumns() []string {
+	return []string{"PostId", "LastReplyAt", "ReplyCount", "Participants"}
+}
+
+func threadToSlice(thread *model.Thread) []interface{} {
+	return []interface{}{
+		thread.PostId,
+		thread.LastReplyAt,
+		thread.ReplyCount,
+		thread.Participants,
+	}
+}
+
+func (s *SqlThreadStore) createIndexesIfNotExists() {
+	s.CreateIndexIfNotExists("idx_threads_last_reply_at", "Threads", "LastReplyAt")
+	s.CreateIndexIfNotExists("idx_threads_post_id", "Threads", "PostId")
```

Comment:
This index is automatically created by both MySQL and Postgres as part of `PostId` being the primary key for the table.  Creating this is unnecessary and will lead to increased storage, slower backups and restores, and slower writes.

Reply:
@mkraft yikes! thanks for noticing this. I'll remove in the next PR related to this feature.